### PR TITLE
feat: restrict batch streaming permissions

### PR DESCRIPTION
**Pull request by mobitar**

### CODE REVIEW

File: packages/snjs/lib/services/component_manager.ts [hunk 1]
```diff
@@ -860,30 +890,37 @@ export class SNComponentManager extends PureService {
...
   }
 
-  handleStreamItemsMessage(component: SNComponent, message: ComponentMessage) {
+  handleStreamItemsMessage(
+    component: SNComponent,
+    message: ComponentMessage
+  ): void {
+    const data = message.data as StreamItemsMessageData;
+    const types = data.content_types
+      .filter((type) => AllowedBatchPermissions.includes(type))
```
Comment (mobitar) [inline packages/snjs/lib/services/component_manager.ts:899]: Main change 1

File: packages/snjs/lib/services/component_manager.ts [hunk 2]
```diff
@@ -1169,11 +1210,16 @@ export class SNComponentManager extends PureService {
...
   }
 
-  handleDeleteItemsMessage(component: SNComponent, message: ComponentMessage) {
+  handleDeleteItemsMessage(
+    component: SNComponent,
+    message: ComponentMessage
+  ): void {
+    const data = message.data as DeleteItemsMessageData;
+    const items = data.items.filter((item) =>
+      AllowedBatchPermissions.includes(item.content_type)
```
Comment (mobitar) [inline packages/snjs/lib/services/component_manager.ts:1219]: Main change 2

**Comment by mobitar** [comment]
Because 90% of the changes are linting, I've commented where you can find the main changes.

File: packages/snjs/lib/services/component_manager.ts (hunk 1 above)
Comment (johnny243) [reply packages/snjs/lib/services/component_manager.ts:899]: `types` is sorted twice:

- on line `900` and line `904`

Comment (mobitar) [reply packages/snjs/lib/services/component_manager.ts:899]: Good catch 👍

**Review by johnny243 — APPROVED** [review]

**Pull request merged**